src/core/config/srp_server.h

```diff
@@ -46,13 +46,37 @@
 #endif
 
 /**
- * @def OPENTHREAD_CONFIG_SRP_SERVER_UDP_PORT
+ * @def OPENTHREAD_CONFIG_SRP_SERVER_UDP_PORT_MIN
  *
- * Specifies the SRP Server UDP port, and use 0 for ephemeral port.
+ * Specifies the min port number in the port range reserved for SRP server.
  *
  */
-#ifndef OPENTHREAD_CONFIG_SRP_SERVER_UDP_PORT
-#define OPENTHREAD_CONFIG_SRP_SERVER_UDP_PORT 0
+#ifndef OPENTHREAD_CONFIG_SRP_SERVER_UDP_PORT_MIN
+#define OPENTHREAD_CONFIG_SRP_SERVER_UDP_PORT_MIN 53535
+#endif
+
+/**
+ * @def OPENTHREAD_CONFIG_SRP_SERVER_UDP_PORT_MAX
+ *
+ * Specifies the max port number in the port range reserved for SRP server.
+ *
+ */
+#ifndef OPENTHREAD_CONFIG_SRP_SERVER_UDP_PORT_MAX
+#define OPENTHREAD_CONFIG_SRP_SERVER_UDP_PORT_MAX 53554
+#endif
+
+/**
+ * @def OPENTHREAD_CONFIG_SRP_SERVER_SAVE_INFO
+ *
+ * Define to 1 to enable SRP server feature to save its own port in non-volatile settings.
+ *
+ * When enabled, the SRP server will save the its port in the non-volatile settings. On a server
+ * restart (e.g., due to a device reset) it will choose a UDP port different from the saved one. The info is written to
+ * Settings after an SRP server successfully starts.
+ *
+ */
+#ifndef OPENTHREAD_CONFIG_SRP_SERVER_SAVE_INFO
+#define OPENTHREAD_CONFIG_SRP_SERVER_SAVE_INFO 1
```

Comment:
```suggestion
#define OPENTHREAD_CONFIG_SRP_SERVER_PORT_SWITCH_ENABLE 1
```
I think the configuration name should be what the feature is, rather than how we implemented it?